roles/osp_observability/templates/prometheus.yml.j2

```diff
@@ -1,18 +1,34 @@
-#jinja2: trim_blocks:False
 global:
   scrape_interval: {{ prometheus_scrape_interval }}
   external_labels:
-    {% if ansible_fqdn %}agent_host: {{ ansible_fqdn }}{% endif %}
-    {% if openstack_cloud_name %}cloud_name: {{ openstack_cloud_name }}{% endif %}
+    {% if ansible_fqdn -%}
+    agent_host: {{ ansible_fqdn }}
+    {% endif -%}
+    {% if openstack_cloud_name -%}
+    cloud_name: {{ openstack_cloud_name }}
+    {% endif %}
 
 scrape_configs:
-{%- for service, nodes in scrape_endpoints.items() %}
+  {% for service, nodes in scrape_endpoints.items() -%}
   - job_name: {{ service }}
     static_configs:
       - targets: {{ nodes }}
-{%- endfor %}
+  {% endfor %}
 
 remote_write:
-{% for url in remote_write -%}
-  - url: {{ url }}
-{% endfor %}
+  {% for name, endpoint in remote_write.items() -%}
+  - url: {{ endpoint.url }}
+    {% if endpoint.basic_user is defined and endpoint.basic_pass is defined -%}
+    basic_auth:
+      username: {{ endpoint.basic_user }}
+      password: {{ endpoint.basic_pass }}
+    {% endif -%}
+    {% if endpoint.url | regex_search('^https://') -%}
+    tls_config:
+      insecure_skip_verify: false
```

Comment:
Any reason we should hard code this to `false` or should it also be parameterized with `| default(false)` ?

Reply:
Good question. Since we will have a documented (and quite simple) procedure for setting up the certificate, there's an argument to be made that we should not support insecure connections, because it shows something was not set up according to the docs. OTOH, I can see having the option available might simplify certain testing scenarios where you don't have easy access to the other side to export the cert.

I will add the parameter.

Reply:
I could also get behind limiting things to matching what we want to support as well, so I'm good with either option.